src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Field.cs

```diff
@@ -12,11 +10,43 @@ public override BoundNode VisitFieldAccess(BoundFieldAccess node)
         {
             BoundExpression rewrittenReceiver = VisitExpression(node.ReceiverOpt);
 
-            // field access of a tuple actually accesses the underlying field.
-            FieldSymbol field = (node.FieldSymbol as TupleFieldSymbol)?.UnderlyingTypleFieldSymbol ??
-                                node.FieldSymbol;
+            var expandedNode = ExpandTupleAccessor(node, rewrittenReceiver);
 
-            return MakeFieldAccess(node.Syntax, rewrittenReceiver, field, node.ConstantValue, node.ResultKind, node.Type, node);
+            return MakeFieldAccess(expandedNode.Syntax, expandedNode.ReceiverOpt, expandedNode.FieldSymbol, expandedNode.ConstantValue, expandedNode.ResultKind, expandedNode.Type, expandedNode);
+        }
+
+        /// <summary>
+        /// Converts access to a tuple instance into access into the underlying ValueTuple(s).
+        ///
+        /// For instance, tuple.Item8
+        /// produces fieldAccess(field=Item1, receiver=fieldAccess(field=Rest, receiver=ValueTuple for tuple))
+        /// </summary>
+        private BoundFieldAccess ExpandTupleAccessor(BoundFieldAccess node, BoundExpression rewrittenReceiver)
+        {
+            var tupleField = node.FieldSymbol as TupleFieldSymbol;
+            if (tupleField == null)
+            {
+                // replace receiver
+                return node.Update(rewrittenReceiver, node.FieldSymbol, node.ConstantValue, node.ResultKind, node.Type);
+            }
+
+            if (tupleField.UnderlyingFieldSymbolChain.Count == 1)
+            {
+                // short tuple: replace receiver and field symbol
+                return node.Update(rewrittenReceiver, tupleField.UnderlyingFieldSymbolChain[0], node.ConstantValue, node.ResultKind, node.Type);
+            }
+
+            // long tuple
+            var nestedField = rewrittenReceiver;
+            var tuple = (TupleTypeSymbol)tupleField.ContainingType;
+            var chainLength = tupleField.UnderlyingFieldSymbolChain.Count;
+            for (var i = 0; i < chainLength; i++)
+            {
+                var nestedFieldType = (i < chainLength - 1) ? tuple.UnderlyingTupleTypeChain[i + 1] : node.Type;
+                nestedField = new BoundFieldAccess(node.Syntax, nestedField, tupleField.UnderlyingFieldSymbolChain[i], node.ConstantValue, node.ResultKind, nestedFieldType);
+            }
```

Comment:
I think, if the code was written without conditional logic inside the loop, it would be more efficient and easier to understand:
```
            var tuple = (TupleTypeSymbol)tupleField.ContainingType;
            int countOfIntermediateFields = tupleField.UnderlyingFieldSymbolChain.Count - 1;
            for (var i = 0; i < countOfIntermediateFields; i++)
            {
                rewrittenReceiver = new BoundFieldAccess(node.Syntax, 
                                                         rewrittenReceiver,
                                                         tupleField.UnderlyingFieldSymbolChain[i],
                                                         ..., ..., 
                                                         tuple.UnderlyingTupleTypeChain[i + 1]);
            }

            return new BoundFieldAccess(node.Syntax, 
                                        rewrittenReceiver,
                                        tupleField.UnderlyingFieldSymbolChain[countOfIntermediateFields],
                                        node.ConstantValue, node.ResultKind, node.Type);
```

Also note that as an added benefit there will be no need to to special case ```tupleField.UnderlyingFieldSymbolChain.Count == 1``` scenario above.